missingno/missingno.py

```diff
@@ -239,6 +239,9 @@ def bar(df, figsize=(24, 10), fontsize=16, labels=None, log=False, color='dimgra
         ax1 = ax
         figsize = None  # for behavioral consistency with other plot types, re-use the given size
 
+    if figsize is not None and orientation != 'bottom':
```

Comment:
This code should go under the `if ax is None:` code block. Otherwise you are manipulating the `figsize` of an already-existing axis, in the case that `orientation != 'bottom" and ax is not None`, which is not the correct behavior. This "smart `figsize` determination"  behavior should only trigger when the user doesn't pass an `ax`.

Also, `figsize = reversed(figsize)` is not a good size setter and is not consistent with what `dendrogram` does. Should be `figsize = (24, (24 + len(df.columns) - 50) * 0.5)` instead.

Reply:
Valid points. Processed both in the next commit.